in_toto/verifylib.py

```diff
@@ -438,12 +451,8 @@ def verify_link_signature_thresholds(layout, chain_link_dict):
 
       # Verify signature and skip invalidly signed links
       try:
-        if securesystemslib.formats.GPG_PUBKEY_SCHEMA.matches(
-            verification_key):
-          keys = [GPGKey.from_dict(verification_key)]
-        elif securesystemslib.formats.KEY_SCHEMA.matches(verification_key):
-          keys = [SSlibKey.from_securesystemslib_key(verification_key)]
-
+        keys = create_key_list_from_key_dict(
+            {authorized_keyid: verification_key})
```

Comment:
Just remembered that we have to be careful when using `from_dict` methods (which you call in `create_key_list_from_key_dict`), as they usually have the side-effect of popping elements from the passed dict.

The reason why we can still use `verification_key["keyid"]` below is that `SslibKey.from_dict` only pops some items put not the keyid and `GPGKey.from_dict` does not pop at all yet. The latter may change with https://github.com/in-toto/securesystemslib/issues/7.

We should  hopefully catch the change with our tests, but it might be worth already updating the code to not use key dicts after they have been passed to `create_key_list_from_key_dict` (and add a note about the side-effect to the function docstring).

Reply:
I found `GPGKey.from_dict` and `SSlibKey.from_securesystemslib_key` both doesn't have any side effects currently.